### DISCUSSION TITLE
Is AppRouterHighlight Still Needed If Using Vercel Integration?

### DISCUSSION THREAD

**Discussion by fischjer4** [highlight/highlight#8076] [Q&A] [answered]
I'm reading through Next[.JS Advanced Config](https://www.highlight.io/docs/getting-started/fullstack-frameworks/next-js/advanced-config#vercel-log-drain) and came across this bit:

> Our API wrappers automatically send logs to Highlight in all runtime environments, but Vercel shuts down its Node.js and Edge processes so quickly that log messages are often lost.
Vercel Log Drain is a reliable way to capture those logs.

Do does this mean that if we are using the Vercel + Highlight integration that we _do not_ need to wrap our endpoints `AppRouterHighlight` since the logs are going to be sent to Highlight regardless via the Vercel Log Drain?

Maybe the key wording here is "in all runtime environments". If you want logs from Local and Staging, then you need to wrap the API routes, but if you only want production, then the Vercel Log Drain is sufficient.

**Comment by deltaepsilon** [comment, marked as answer]
The Vercel + Highlight Log Drain integration works without wrapping API endpoints.

You'll want to wrap the endpoints for Errors and Traces coverage.

AFAIK, the runtime environments all act the same, but I haven't done any detailed testing there, so I'm curious if you find something different.
